src/fft.rs

```diff
@@ -0,0 +1,165 @@
+//! Implementation of Real valued FFT
+use num_traits::Float;
+
+use crate::fft_64;
+use crate::Direction;
+
+fn compute_twiddle_factors<T: Float>(big_n: usize) -> (Vec<T>, Vec<T>) {
+    let half_n = big_n / 2;
+    let mut real_parts = Vec::with_capacity(half_n);
+    let mut imag_parts = Vec::with_capacity(half_n);
+    let two: T = T::from(2.0).unwrap();
+    let pi: T = T::from(std::f64::consts::PI).unwrap();
+
+    for k in 0..half_n {
+        let angle = -two * pi * T::from(k).unwrap() / T::from(big_n).unwrap();
+        real_parts.push(angle.cos());
+        imag_parts.push(angle.sin());
+    }
+
+    (real_parts, imag_parts)
+}
+
+/// Implementation of Real-Valued FFT
+pub fn r2c_fft_f64(input_re: &[f64]) -> (Vec<f64>, Vec<f64>) {
```

Comment:
Any chance this could be non-allocating and take &mut outputs?

Reply:
Let me now if the checked in fix will suffice